### ISSUE TITLE
FreeBSD agent and installscript

### ISSUE THREAD

**Issue by NetBLOKS** [amidaware/tacticalrmm#1525]
**Is your feature request related to a problem? Please describe.**
We need FreeBSD as agent support.

**Describe the solution you'd like**
A compatible TRMM Agent and install script including Meshagent (which is FreeBSD compatible).
Just like the Linux agent and installscript.

**Describe alternatives you've considered**
No alternatives. Remotesessions via VPN and SSH at the moment. But no inventory and script management.

**Additional context**
This would really set TRMM apart, since there are very few RMM solutions that include FreeBSD agents.

**Comment by LPJon** [comment]
Yes!, I would love to see this as well. Opensource firewall management is off the board as long as this is not supported. Would any funding help move this along?

**Comment by NetBLOKS** [comment]
Jup, would also add to the bounty/ funding!

**Comment by LPJon** [comment]
Well, I'm willing to pay specifically for this feature, I just need a number..lol

**Comment by dinger1986** [comment]
just looking through issues on https://github.com/amidaware/rmmagent and found this also from @LPJon https://github.com/amidaware/rmmagent/issues/16

Appears theres dependencies required, have these been satisfied now?

**Comment by LPJon** [comment]
I don't believe so....but that was only for TacticalRMM to read information from the client. Has nothing to do with Meshcentral using the "Take Control" function of TacticalRMM. I wasn't asking for all the info data in TacticalRMM. Just the ability to use Meshcentral to connect to the OPNSense (FreeBSD) system. I don't particularly need to run scripts or see info in TactcalRMM's interface. I want to be able to connect to the system through TacticalRMM's interface and use meshcentral which already supports it.

**Comment by dinger1986** [comment]
if its supported in anyway other users will expect features which exist on other agents so they need implemented at least to the same level as the linux agent is at now.

**Comment by LPJon** [comment]
@dinger1986 Yeah, I can see that. It's just a royal pain to have to maintain two servers just to access FreeBSD based clients. This would make a huge leap for the project as firewalls are a big deal. Maybe a simple feature to only install the MeshAgent and register the agent with TacticalRMM would work. As in like a type of TacticalRMM Agent Install. Then have that type of install only allow the Take Control option and leave everything else. You should also be able to collect most of the data already in TacticalRMM from the MeshCentral instance if that nessisary. It already tracks quite a bit of info when it's installed and you can get it in JSON from the Mesh CLI. Running scripts on FreeBSD isn't the issue. It's getting system info from the "proc" partition. That's probably never going to be supported with GO. We really want this to happen. Even if it's an exclusive "Turn me on" option where we have to turn it on via a UI setting. Like I said, I need this bad enough that I will pay for it. We really need an all-in-one option which is why we chose TactcalRMM. It really is a great project. I have really enjoyed using it and it makes my life so much easier in a lot of other ways. This would really add to that ease of use.
